pkg/vm/stackitem/serialization.go

```diff
@@ -9,42 +9,49 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/io"
 )
 
-// SerializeItem encodes given Item into the byte slice.
-func SerializeItem(item Item) ([]byte, error) {
+// SerializeItem encodes given Item into the byte slice. Int in return values
+// corresponds to the overall size of the underlying primitive types.
+func SerializeItem(item Item) ([]byte, int, error) {
 	w := io.NewBufBinWriter()
-	EncodeBinaryStackItem(item, w.BinWriter)
+	size := EncodeBinaryStackItem(item, w.BinWriter)
 	if w.Err != nil {
-		return nil, w.Err
+		return nil, 0, w.Err
 	}
-	return w.Bytes(), nil
+	return w.Bytes(), size, nil
```

Comment:
But the size is just a `len(w.Bytes())`? I don't think we need to return it explicitly.

Reply:
Not exactly. We have to count it separately because the size shouldn't take into account stackitem types. And I see two options here: either modify `serializeItemTo` in order to return the overall stackitem size without types size or make a separate function for `runtimeNotify` which counts the size of stackitem without types. See the C# implementation: https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/ApplicationEngine.Runtime.cs#L55

Reply:
neo-project/neo#2069

Reply:
Fixed.

Reply:
What's the maximum size we should use for notification? `stackitem.MaxSize` is too big.

Reply:
`MaxNotificationSize`.